sqlglot/dialects/tsql.py

```diff
@@ -42,11 +50,34 @@ def _format_time(args):
     return _format_time
 
 
+def parse_format(args):
+    fmt = list_get(args, 1)
+    number_fmt = not re.search(DATE_FMT_RE, fmt.this) or fmt.this in TRANSPILE_SAFE_NUMBER_FMT
+    if number_fmt:
+        return exp.NumberToStr(this=list_get(args, 0), format=fmt)
+    else:
+        return exp.TimeToStr(
+            this=list_get(args, 0), format=exp.Literal.string(format_time(fmt.name, TSQL.time_mapping))
+        )
+
+
 def generate_date_delta_with_unit_sql(self, e):
     func = "DATEADD" if isinstance(e, exp.DateAdd) else "DATEDIFF"
     return f"{func}({self.format_args(e.text('unit'), e.expression, e.this)})"
 
 
+def generate_format_sql(self, e):
+    fmt = (
+        e.args.get("format")
+        if isinstance(e, exp.NumberToStr)
+        else exp.Literal.string(
+            format_time(e.args.get("format").name, {v: k.lower() for k, v in TSQL.time_mapping.items()})
+        )
+    )
+    print(self.sql(e, "format"))
+    return f"FORMAT({self.format_args(e.this, fmt)})"
+
+
```

Comment:
You can simplify `e.args.get("format")` to `e.args["format"]` (also: you forgot the print).